### PR TITLE
EDLY 7202 Register LMS user on mixpanel

### PR DESCRIPTION
**Pull request by Waleed-Mujahid**
## Description
This PR registers an LMS user on mixpanel.

## Ticket
https://edlyio.atlassian.net/browse/EDLY-7202

### CODE REVIEW

File: openedx/core/djangoapps/user_authn/views/register.py [hunk 1]
```diff
@@ -230,6 +231,8 @@ def create_account_with_params(request, params):
 
     edly_access_user = create_edly_access_role(request, user)
     create_learner_link_with_permission_groups(edly_access_user)
+    sub_org = get_edly_sub_org_from_request(request)
+    register_user_on_mixpanel(request.user, sub_org)
```
Comment (Anas-hameed) [inline openedx/core/djangoapps/user_authn/views/register.py:235]: Please move this code to edly-panel-app user route.

File: cms/templates/js/course-outline.underscore [hunk 2]
```diff
@@ -147,7 +147,7 @@ if (is_proctored_exam) {
                                 {
                                     'service': 'studio',
                                     'screen_name': 'course_outline',
-                                    'organization': <%- platform_name %>,
+                                    'organization': '${ platform_name }',
```
Comment (Anas-hameed) [inline cms/templates/js/course-outline.underscore:150]: Can you explain this change?

File: cms/templates/js/certificate-web-preview.underscore [hunk 3]
```diff
@@ -7,7 +7,7 @@
 <a href=<%- certificate_web_view_url %> class="button preview-certificate-link" rel="noopener" target="_blank">
     <%- gettext("Preview Certificate") %>
 </a>
-<button class="button activate-cert" onclick="mixpanel.track('certificate_activate',{'service':'studio', 'screen_name':'certificates', 'organization':<%-platform_name%>})">
+<button class="button activate-cert" onclick="mixpanel.track('certificate_activate', {'service': 'studio', 'screen_name': 'certificates', 'organization': '${ platform_name }'})">
```
Comment (Anas-hameed) [inline cms/templates/js/certificate-web-preview.underscore:10]: Can we make this change from themes?
https://edx.readthedocs.io/projects/edx-installing-configuring-and-running/en/latest/configuration/changing_appearance/theming/create_theme.html

File: openedx/features/edly/context_processor.py [hunk 4]
```diff
@@ -69,6 +72,7 @@ def edly_app_context(request):  # pylint: disable=unused-argument
             'ga_id': configuration_helpers.get_value('GA_ID'),
             'mixpanel_project_token': settings.MIXPANEL_PROJECT_TOKEN,
             'usetiful_token': settings.USETIFUL_TOKEN,
+            'email': get_email_and_register_mixpanel(request),
```
Comment (Anas-hameed) [inline openedx/features/edly/context_processor.py:75]: Why this change is in context processor?

File: openedx/features/edly/utils.py [hunk 5]
```diff
@@ -899,3 +901,51 @@ def get_enrolled_learners_count(course_run_ids):
...
+
+    lms_role = 'learner'
+    if user.is_superuser:
+        lms_role = 'super_admin'
+    elif auth.user_has_role(user, GlobalCourseCreatorRole(edly_sub_org.slug), False):
+        lms_role = 'staff'
+    elif auth.user_has_role(user, CourseCreatorRole(), False):
+        lms_role = 'course_creator'
+
+    return lms_role
```
Comment (Anas-hameed) [inline openedx/features/edly/utils.py:921]: Don't we have this similar util function already available?

File: openedx/features/edly/utils.py [hunk 6]
```diff
@@ -899,3 +901,51 @@ def get_enrolled_learners_count(course_run_ids):
...
+    Build the user properties dictionary to send to Mixpanel.
+    """
+    return {
+        '$name': user.username,
+        '$email': user.email,
+        'department': get_user_lms_role(user, edly_sub_org),
+    }
+
+
+def get_email_and_register_mixpanel(request):
```
Comment (Anas-hameed) [inline openedx/features/edly/utils.py:935]: nit: Sigle responsibility -> a function should not do more than one thing.

**Review by Anas-hameed — CHANGES_REQUESTED** [review]

File: cms/templates/js/course-outline.underscore (hunk 2 above)
Comment (Waleed-Mujahid) [reply cms/templates/js/course-outline.underscore:150]: This syntax was causing some issue. The event was not being triggered previously

File: cms/templates/js/certificate-web-preview.underscore (hunk 3 above)
Comment (Waleed-Mujahid) [reply cms/templates/js/certificate-web-preview.underscore:10]: Ok sure but should I remove the existing code from this file as well? Because there were already changes in it

**Review by Anas-hameed — APPROVED** [review]

File: openedx/features/edly/utils.py [hunk 7]
```diff
@@ -899,3 +901,50 @@ def get_enrolled_learners_count(course_run_ids):
...
+def get_user_lms_role(user, edly_sub_org):
+    """
+    Get the lms role for user
+    """
+    # Prevent a circular import.
+    from student import auth
+
+    lms_role = 'learner'
+    if user.is_superuser:
+        lms_role = 'super_admin'
```
[the remark this answers is not in the document]
Comment (Waleed-Mujahid) [reply openedx/features/edly/utils.py]: I have created a new enum for it

**Review by muhammadali286 — APPROVED** [review]
